### ISSUE TITLE
The manual has missing cross-references

### ISSUE THREAD

**Issue by rptb1** [Ravenbrook/mps#158]
The MPS manual has some missing cross-references.  References to the build instructions, such as at https://memory-pool-system.readthedocs.io/en/latest/topic/porting.html#update-the-documentation show as "guide-build" without a link.

A clean build of the manual at e179634de028b2db090ee6a2358391155102b74f on Ubuntu 22 generates these warnings:
```
/home/rb/git/mps/manual/source/code-index.rst:19: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/glossary/c.rst:567: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/guide/lang.rst:18: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/guide/overview.rst:58: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/guide/overview.rst:145: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/release.rst:918: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/release.rst:921: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/topic/porting.rst:6: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/topic/porting.rst:244: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/topic/porting.rst:313: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
/home/rb/git/mps/manual/source/topic/scanning.rst:559: WARNING: Unparseable C cross-reference: 'void *'
Invalid C declaration: Expected identifier in nested name, got keyword: void [error at 4]
  void *
  ----^
/home/rb/git/mps/manual/source/topic/telemetry.rst:75: WARNING: Failed to create a cross reference. A title or caption not found: guide-build
```

Some kind of anchor has gone missing from [build.txt](https://github.com/Ravenbrook/mps/blob/master/manual/build.txt).

**Comment by rptb1** [comment]
This was not found earlier because `make -C manual clean` is broken. It assumes incorrectly that the shell run by make has extended globbing (i.e. `{}` expansion) https://github.com/Ravenbrook/mps/blob/e179634de028b2db090ee6a2358391155102b74f/manual/Makefile#L47

Introduced in 7a90c63e9367337fe322887a95beef648a69b3f3.

**Comment by rptb1** [comment]
This error was introduced in 2ef000acda7ad33fb998a5845ab0650e5922e879 as part of #105 .  Fixing the markup of build.txt to be a correct standalone reStructuredText document changed its initial heading to a document title, preventing Sphinx from recognizing as a section heading, a valid target for this anchor https://github.com/Ravenbrook/mps/blob/e179634de028b2db090ee6a2358391155102b74f/manual/source/guide/build.rst?plain=1#L8-L10
which is how build.txt gets included in the manual.

**Comment by rptb1** [comment]
This might be best solved by generalizing the [MPS Sphinx extension](https://github.com/Ravenbrook/mps/tree/e179634de028b2db090ee6a2358391155102b74f/manual/source/extensions/mps) to convert more than just the design documents.  I have wondered elsewhere whether we might want to include procedures and other standalone reStructuredText documents in the manual.

**Comment by rptb1** [comment]
Read the Docs CI "passed" even with missing cross-references.  That's no good.  We should enable [`fail_on_warning`](https://docs.readthedocs.io/en/stable/config-file/v2.html#sphinx)

**Comment by rptb1** [comment]
> Read the Docs CI "passed" even with missing cross-references. That's no good. We should enable [`fail_on_warning`](https://docs.readthedocs.io/en/stable/config-file/v2.html#sphinx)

Fixed in #159 https://github.com/Ravenbrook/mps/pull/159/commits/d7858985cf84da511f81b9d72f3a1ce3c4f8c477
